### PR TITLE
[206] Import Export

### PR DESCRIPTION
**Pull request by orichalque**
#206

### CODE REVIEW

**Comment by orichalque** [comment]
@RaphaelPage0110 not sure that the import / export buttons are well positioned though. 
Should we put it in a common dropdown menu with the Save button ? I think the save button should remain quick to use so I dont think so :s 
Wdyt ?

**Comment by RaphaelPage0110** [comment]
> @RaphaelPage0110 not sure that the import / export buttons are well positioned though. Should we put it in a common dropdown menu with the Save button ? I think the save button should remain quick to use so I dont think so :s Wdyt ?
> 
> Maybe it could be in a dropdown submenu along with the "Save gameplay" data option ?
> 
> ATM: image
> 
> Proposition 1: image

I agree, I don't think putting it in the About section is a good idea.  Your proposition 1 seems good. The only concern I have is that the Options submenu is becoming quite large. But I think it's ok for now!

File: js/main.js [hunk 1]
```diff
@@ -146,57 +146,162 @@ new Vue({
...
+            document.body.appendChild(element);          
+            element.click();          
+            document.body.removeChild(element);
+        },
+        importData: function() {
+            $("#import-input").trigger("click")
+        },
+        readDataFile: async function(event) {
+            const file = event.target.files.item(0)
+            //const text = await file.text();
```
Comment (RaphaelPage0110) [inline js/main.js:200]: you can probably remove this

**Review by RaphaelPage0110 — CHANGES_REQUESTED** [review]
- The name of the export file is currently just 'save.json'. Maybe you could name it something a little bit more explicit like "gloomhaven_deckbuilder_export" or something?
- The special modifiers are not exported. Is that normal?

Very good job otherwise, this is awesome!

**Comment by RaphaelPage0110** [comment]
> **A reviewer left a comment:**
> 
> Export wont work if no class is selected; That's pretty obvious, but I suppose a red dialog should be opened to log this event.
> Browser metadata
> 
> [Open Deploy Preview](https://deploy-preview-208--gloomhaven-dev.netlify.app/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=61dd9d94c91e83000773c56e&commentID=61ded4e9a0f5b7218e1eb23a&resolution=resolved)

Good idea too

File: js/main.js (hunk 1 above)
Comment (orichalque) [reply js/main.js:200]: yup

**Comment by orichalque** [comment]
> * The name of the export file is currently just 'save.json'. Maybe you could name it something a little bit more explicit like "gloomhaven_deckbuilder_export" or something?

Good point, will do 

> * The special modifiers are not exported. Is that normal?

No it's not. I'm checking this out

**Comment by RaphaelPage0110** [comment]
I have the following error in the console when importing data : 
image

**Comment by RaphaelPage0110** [comment]
Also the file name is still save.json :)

**Comment by orichalque** [comment]
> Also the file name is still save.json :)

yeah thats because im not done with these changes

**Review by RaphaelPage0110 — APPROVED** [review]

**Comment by RaphaelPage0110** [comment]
It's perfect! You can merge it if you are done :)

**Pull request merged**